chainercv/links/ssd/ssd.py

```diff
@@ -0,0 +1,278 @@
+                cy = (v + 0.5) * self.steps[k]
+
+                s = self.sizes[k]
+                self._default_bbox.append((cx, cy, s, s))
+
+                s = np.sqrt(self.sizes[k] * self.sizes[k + 1])
+                self._default_bbox.append((cx, cy, s, s))
+
+                s = self.sizes[k]
+                for ar in self.aspect_ratios[k]:
+                    self._default_bbox.append(
+                        (cx, cy, s * np.sqrt(ar), s / np.sqrt(ar)))
+                    self._default_bbox.append(
+                        (cx, cy, s / np.sqrt(ar), s * np.sqrt(ar)))
+        self._default_bbox = np.stack(self._default_bbox)
+
+    def to_cpu(self):
+        super(SSD, self).to_cpu()
+        self._default_bbox = chainer.cuda.to_cpu(self._default_bbox)
+
+    def to_gpu(self):
+        super(SSD, self).to_gpu()
+        self._default_bbox = chainer.cuda.to_gpu(self._default_bbox)
+
+    def features(self, x):
+        """Compute feature maps from a batch of images.
+
+        This is a virtual method.
+        The inheriting class must implement this method.
+
+        Args:
+            x (ndarray): An array holding a batch of images.
+                The images are preprocessed by :meth:`prepare` if needed.
+
+        Returns:
+            list of chainer.Variable:
+            Each variable contains a feature map.
+        """
+        raise NotImplementedError
+
+    def _multibox(self, xs):
+        ys_loc = list()
+        ys_conf = list()
+        for i, x in enumerate(xs):
+            loc = self.loc[i](x)
+            loc = F.transpose(loc, (0, 2, 3, 1))
+            loc = F.reshape(loc, (loc.shape[0], -1, 4))
+            ys_loc.append(loc)
+
+            conf = self.conf[i](x)
+            conf = F.transpose(conf, (0, 2, 3, 1))
+            conf = F.reshape(
+                conf, (conf.shape[0], -1, self.n_class + 1))
+            ys_conf.append(conf)
+
+        y_loc = F.concat(ys_loc, axis=1)
+        y_conf = F.concat(ys_conf, axis=1)
+
+        return y_loc, y_conf
+
+    def __call__(self, x):
+        """Compute localization and classification from a batch of images.
+
+        This method computes two variables, :obj:`loc` and :obj:`conf`.
+        :meth:`_decode` converts these variables to prediction.
+        These variables are also used in training SSD.
+
+        Args:
+            x (chainer.Variable): A variable holding a batch of images.
+                The images are preprocessed by :meth:`prepare` if needed.
+
+        Returns:
+            tuple of chainer.Variable:
+            This method returns two variables, :obj:`loc` and :obj:`conf`.
+
+            * **loc**: A variable of float arrays of shape :math:`(K, 4)`, \
+                where :math:`K` is the number of default bounding boxes.
+            * **conf**: A variable of float arrays of shape \
+                :math:`(K, n\_class + 1)`, \
+                where :math:`K` is the number of default bounding boxes.
```

Comment:
Looking at your tests, it seems that the shape of outputs are batched (i.e. (1, K, 4)).
Is the documentation outdated?